build-system/tasks/e2e/describes-e2e.js

```diff
@@ -510,6 +510,16 @@ class EndToEndFixture {
 
     try {
       await setUpTest(env, this.spec);
+      if (env.environment === AmpdocEnvironment.VIEWER_DEMO) {
+        env.receivedMessages = await controller.evaluate(() => {
+          // The viewer.html file will launch 7 test viewers, only one of which is the requested url.
+          // TODO(gh/amphtml/28200): only load the one viewer.
+          const viewer = window.parent.allViewers.find((v) =>
+            v.id.includes('dynamic')
+          );
+          return viewer.receivedMessages;
+        });
+      }
```

Comment:
Nit: Can we move this into `setUpTest()`?

Reply:
We can, but we'd lose out on a useful property. Right now all assignments to `env` occur in the same place. I'd rather leave this ugliness here and we can probably come up with a better separation of logic if other e2e cases also require adding to env after `setUpTest`.

Reply:
Hm, makes sense. What about moving it to the end of the function? E.g. `// Set env props that require the fixture to be set up.`.

It'll also avoid triggering the `catch` block in case `controller.evaluate()` throws.

Reply:
Good thinking. Done

Reply:
I ended up moving it back to the try block. While an error will trigger the `catch` which is meh (not broken, but will fail slower), a naive placement below the try/catch causes its own issues.  Specifically that `this.setup()` would cause the controller block to be called twice if it passed on a second retry. The second call would use the failed first controller yielding a broken call.

This could be fixed by a `return` after the `await this.setup()` in the catch block, but that feels counterintuitive as well.